app/src/main/java/com/android/calendar/settings/CalendarDataStore.kt

```diff
@@ -51,18 +63,47 @@ class CalendarDataStore(activity: FragmentActivity, calendarId: Long) : Preferen
         }
     }
 
+    private fun mapPreferenceKeyToDatabaseKeyForTaskList(key: String?): String {
+        return when (key) {
+            CalendarPreferences.SYNCHRONIZE_KEY -> DmfsOpenTasksContract.TaskLists.COLUMN_SYNC_ENABLE
+            CalendarPreferences.VISIBLE_KEY -> DmfsOpenTasksContract.TaskLists.COLUMN_VISIBLE
+            CalendarPreferences.COLOR_KEY -> DmfsOpenTasksContract.TaskLists.COLUMN_COLOR
+            CalendarPreferences.DISPLAY_NAME_KEY -> DmfsOpenTasksContract.TaskLists.COLUMN_NAME
+            else -> throw UnsupportedOperationException("unsupported preference key")
+        }
+    }
+
     override fun putBoolean(key: String?, value: Boolean) {
-        val databaseKey = mapPreferenceKeyToDatabaseKey(key)
+        val databaseKey: String
```

Comment:
intent is always populated

Reply:
can you give some reasoning why it is always populated?

Reply:
cause the intent is connected to the event, and the event could be an "event" or a "task"

Reply:
so you say it can't be null in that specific case or in general?
Just wondering because it is checked here: https://github.com/Etar-Group/Etar-Calendar/blob/master/app/src/main/java/com/android/calendar/EventInfoActivity.java#L88

Reply:
when we check this the intent has already been initializated